tests/models/test_models_unet_2d_condition.py

```diff
@@ -606,6 +606,22 @@ def test_pickle(self):
 
         assert (sample - sample_copy).abs().max() < 1e-4
 
+    def test_asymmetrical_unet(self):
```

Comment:
nice test!

Reply:
``` # 4. mid
        if self.mid_block is not None:
            sample = self.mid_block(
                sample,
                emb,
                encoder_hidden_states=encoder_hidden_states,
                attention_mask=attention_mask,
                cross_attention_kwargs=cross_attention_kwargs,
                encoder_attention_mask=encoder_attention_mask,
            )```
            These lines make the *args, **kwargs necessary. Unless I should add a the ``` has_attention``` attribute and work off of that, like the Downsampling and upsampling blocks?